File: Remove_Duplicates_from_Sorted_Array/remove_duplicates_from_sorted_array.py
from typing import Optional, List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        """
        Do not return anything, modify nums in-place instead.
        """
        write_index = 1  # Index to write the next unique element

        for read_index in range(1, len(nums)):
            if nums[read_index] != nums[write_index - 1]:
                nums[write_index] = nums[read_index]
                write_index += 1
        
        return write_index
    
class Solution_Others:
    def removeDuplicates(self, nums: List[int]) -> int:
        """
        Do not return anything, modify nums in-place instead.
        """
        write_index = 1  # Index to write the next unique element

        for read_index in range(1, len(nums)):
            if nums[read_index-1] < nums[read_index]:
                nums[write_index] = nums[read_index]
                write_index += 1
        
        return write_index

# Example usage:
if __name__ == "__main__":
    solution = Solution_Others()
    nums = [0,0,1,1,1,2,2,3,3,4]
    length = solution.removeDuplicates(nums)
    print("Length of array after removing duplicates:", length)
    print("Modified array:", nums[:length])  # Output: [1, 2]

File: Remove_Duplicates_from_Sorted_Array/remove_duplicates_from_sorted_array.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution{
    public:
        int removeDuplicates(vector<int>& nums) {
            cout << "nums address: "<< &nums << endl;
            if (nums.empty()) return 0;
            int uniqueIndex = 1; // Start from the second element
            
            for (int i = 1; i < nums.size(); i++) {
                if (nums[i] != nums[i - 1]) {
                    nums[uniqueIndex] = nums[i];
                    uniqueIndex++;
                }
            }
            return uniqueIndex;
        }
};

class Solution_Others{
    public:
        int removeDuplicates(vector<int>& nums) {
            int j = 1;
            for (int i = 1; i < nums.size(); i++) {
                if (nums[i] > nums[i - 1]) {
                    nums[j] = nums[i];
                    j++;
                }
            }
            return j;
        }
    };

int main() {
    Solution solution;
    vector<int> nums = {0,0,1,1,1,2,2,3,3,4};
    cout << "Original array address: "<< &nums << endl;
    int newLength = solution.removeDuplicates(nums);
    cout << "New length: " << newLength << endl;
    cout << "Array after removing duplicates: ";
    for (int i = 0; i < newLength; i++) {
        cout << nums[i] << " ";
    }
    return 0;
}